cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -769,46 +792,114 @@ private class NormalCastingNodeExt extends CastingNodeExt, NormalNodeExt {
 
 private class ReadStoreNodeExt extends CastingNodeExt, TReadStoreNode {
   private DataFlowCall call;
-  private Node node1;
-  private Node node2;
+  private ArgumentNode arg;
   private Content f1;
-  private Content f2;
 
-  ReadStoreNodeExt() { this = TReadStoreNode(call, node1, node2, f1, f2) }
+  ReadStoreNodeExt() { this = TReadStoreNode(call, arg, f1) }
 
   override Node getNode() { none() }
 
   override DataFlowType getErasedNodeTypeBound() { result = f1.getType() }
 
-  override DataFlowCallable getEnclosingCallable() { result = node1.getEnclosingCallable() }
+  override DataFlowCallable getEnclosingCallable() { result = arg.getEnclosingCallable() }
+
+  override predicate isCand1(Configuration config) { nodeCand1(arg, config) }
+
+  override string toString() { result = "(inside) " + call.toString() + " [read " + f1 + "]" }
 
-  override predicate isCand1(Configuration config) {
-    nodeCand1(node1, config) and nodeCand1(node2, config)
+  override predicate hasLocationInfo(
+    string filepath, int startline, int startcolumn, int endline, int endcolumn
+  ) {
+    call.getLocation().hasLocationInfo(filepath, startline, startcolumn, endline, endcolumn)
```

Comment:
Change `call` to `arg`?

Reply:
OK (though that wasn't changed on this PR).

Reply:
I know. It just seemed neater and matches up with the two new cases.